packages/flutter_tools/lib/src/hot.dart

```diff
@@ -493,6 +493,15 @@ class HotRunner extends ResidentRunner {
       printError('Hot reload failed:\ncode = $errorCode\nmessage = $errorMessage\n$st');
       return false;
     }
+    // Reload the isolate.
+    await currentView.uiIsolate.reload();
+    // Check if the isolate is paused.
+    final ServiceEvent pauseEvent = currentView.uiIsolate.pauseEvent;
+    if ((pauseEvent != null) && (pauseEvent.isPauseEvent)) {
+      // Isolate is paused. Stop here.
+      printStatus('Skipping reassemble because isolate is paused.');
```

Comment:
This should probably be a printTrace to match the `Reassembling application` message below. If this should be user-facing, we may want to update the message to make it clearer to the user what's happening.